### ISSUE TITLE
build vicuna-v1-7b tvm error

### ISSUE THREAD

**Issue by cc11235** [mlc-ai/mlc-llm#182]
build code:
`python build.py --model vicuna-v1-7b --quantization q4f16_0 --target android --max-seq-len 768`

error:

```
Using model path dist/models/vicuna-v1-7b
Load cached module from dist/vicuna-v1-7b-q4f16_0/mod_cache_before_build_android.pkl and skip tracing. You can use --use-cache=0 to retrace
19 static functions: [I.GlobalVar("fused_decode_fused_matmul2_add"), I.GlobalVar("take_decode"), I.GlobalVar("reshape2"), I.GlobalVar("fused_transpose4_reshape4"), I.GlobalVar("slice1"), I.GlobalVar("reshape"), I.GlobalVar("rotary_embedding1"), I.GlobalVar("fused_reshape2_squeeze"), I.GlobalVar("fused_decode2_fused_matmul5_add"), I.GlobalVar("divide2"), I.GlobalVar("fused_decode1_fused_matmul4_silu"), I.GlobalVar("fused_decode3_fused_matmul6_cast2"), I.GlobalVar("reshape1"), I.GlobalVar("softmax2"), I.GlobalVar("transpose2"), I.GlobalVar("fused_decode1_fused_matmul4_multiply"), I.GlobalVar("fused_decode_matmul2"), I.GlobalVar("squeeze"), I.GlobalVar("rms_norm1")]
26 dynamic functions: [I.GlobalVar("extend_te"), I.GlobalVar("transpose3"), I.GlobalVar("rotary_embedding"), I.GlobalVar("fused_softmax1_cast4"), I.GlobalVar("rms_norm"), I.GlobalVar("fused_decode6_fused_NT_matmul4_add1"), I.GlobalVar("matmul8"), I.GlobalVar("fused_NT_matmul_divide_maximum_minimum_cast"), I.GlobalVar("reshape8"), I.GlobalVar("reshape3"), I.GlobalVar("fused_decode5_fused_NT_matmul3_multiply1"), I.GlobalVar("squeeze1"), I.GlobalVar("reshape7"), I.GlobalVar("fused_decode4_fused_NT_matmul1_add1"), I.GlobalVar("slice"), I.GlobalVar("fused_NT_matmul2_divide1_maximum1_minimum1_cast3"), I.GlobalVar("matmul3"), I.GlobalVar("full"), I.GlobalVar("fused_min_max_triu_te_broadcast_to"), I.GlobalVar("fused_decode4_NT_matmul1"), I.GlobalVar("reshape6"), I.GlobalVar("transpose6"), I.GlobalVar("fused_decode5_fused_NT_matmul3_silu1"), I.GlobalVar("reshape5"), I.GlobalVar("fused_softmax_cast1"), I.GlobalVar("take_decode1")]
Dump static shape TIR to dist/vicuna-v1-7b-q4f16_0/debug/mod_tir_static.py
Dump dynamic shape TIR to dist/vicuna-v1-7b-q4f16_0/debug/mod_tir_dynamic.py
- Dispatch to pre-scheduled op: fused_softmax1_cast4
- Dispatch to pre-scheduled op: rms_norm
- Dispatch to pre-scheduled op: matmul8
- Dispatch to pre-scheduled op: fused_NT_matmul_divide_maximum_minimum_cast
- Dispatch to pre-scheduled op: fused_NT_matmul2_divide1_maximum1_minimum1_cast3
- Dispatch to pre-scheduled op: matmul3
- Dispatch to pre-scheduled op: fused_min_max_triu_te_broadcast_to
- Dispatch to pre-scheduled op: fused_softmax_cast1
[01:12:19] /home/notebook/code/personal/project/llm_android/relax/relax/src/runtime/opencl/opencl_device_api.cc:459: Warning: No OpenCL platform matched given existing options ...
Traceback (most recent call last):
  File "/home/notebook/code/personal/project/llm_android/mlc-llm/build.py", line 330, in <module>
    build(mod, ARGS)
  File "/home/notebook/code/personal/project/llm_android/mlc-llm/build.py", line 265, in build
    ex = relax.build(mod_deploy, args.target, system_lib=args.system_lib)
  File "/home/notebook/code/personal/project/llm_android/relax/relax/python/tvm/relax/vm_build.py", line 337, in build
    return _vmlink(builder, target, tir_mod, ext_libs, params, system_lib=system_lib)
  File "/home/notebook/code/personal/project/llm_android/relax/relax/python/tvm/relax/vm_build.py", line 242, in _vmlink
    lib = tvm.build(
  File "/home/notebook/code/personal/project/llm_android/relax/relax/python/tvm/driver/build_module.py", line 281, in build
    rt_mod_host = _driver_ffi.tir_to_runtime(annotated_mods, target_host)
  File "/home/notebook/code/personal/project/llm_android/relax/relax/python/tvm/_ffi/_ctypes/packed_func.py", line 238, in __call__
    raise get_last_ffi_error()
tvm._ffi.base.TVMError: Traceback (most recent call last):
  8: TVMFuncCall
  7: tvm::runtime::PackedFuncObj::Extractor<tvm::runtime::PackedFuncSubObj<tvm::runtime::TypedPackedFunc<tvm::runtime::Module (tvm::runtime::Map<tvm::Target, tvm::IRModule, void, void> const&, tvm::Target)>::AssignTypedLambda<tvm::{lambda(tvm::runtime::Map<tvm::Target, tvm::IRModule, void, void> const&, tvm::Target)#6}>(tvm::{lambda(tvm::runtime::Map<tvm::Target, tvm::IRModule, void, void> const&, tvm::Target)#6}, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >)::{lambda(tvm::runtime::TVMArgs const&, tvm::runtime::TVMRetValue*)#1}> >::Call(tvm::runtime::PackedFuncObj const*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, tvm::runtime::TVMRetValue)
  6: tvm::TIRToRuntime(tvm::runtime::Map<tvm::Target, tvm::IRModule, void, void> const&, tvm::Target const&)
  5: tvm::codegen::Build(tvm::IRModule, tvm::Target)
  4: tvm::runtime::PackedFuncObj::Extractor<tvm::runtime::PackedFuncSubObj<tvm::runtime::TypedPackedFunc<tvm::runtime::Module (tvm::IRModule, tvm::Target)>::AssignTypedLambda<tvm::codegen::{lambda(tvm::IRModule, tvm::Target)#1}>(tvm::codegen::{lambda(tvm::IRModule, tvm::Target)#1}, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >)::{lambda(tvm::runtime::TVMArgs const&, tvm::runtime::TVMRetValue*)#1}> >::Call(tvm::runtime::PackedFuncObj const*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, tvm::runtime::TVMRetValue)
  3: tvm::codegen::LLVMModuleNode::Init(tvm::IRModule const&, tvm::Target const&)
  2: tvm::codegen::CodeGenCPU::Finish()
  1: tvm::codegen::CodeGenLLVM::Finish()
  0: tvm::codegen::CodeGenLLVM::Verify() const
  File "/home/notebook/code/personal/project/llm_android/relax/relax/src/target/llvm/codegen_llvm.cc", line 318
TVMError: LLVM module verification failed with the following errors: 
!dbg attachment points at wrong subprogram for function
!2441 = distinct !DISubprogram(name: "main.tir", scope: !1, file: !1, type: !4, spFlags: DISPFlagLocalToUnit | DISPFlagDefinition | DISPFlagOptimized, unit: !0, retainedNodes: !2442)
i32 (i8*, i32*, i32, i8*, i32*, i8*)* @main
  %6 = alloca i8*, !dbg !2450
!2450 = !DILocation(line: 0, scope: !2451)
!2451 = distinct !DISubprogram(name: "main.tir", scope: !1, file: !1, type: !4, spFlags: DISPFlagLocalToUnit | DISPFlagDefinition | DISPFlagOptimized, unit: !0, retainedNodes: !2452)
!2451 = distinct !DISubprogram(name: "main.tir", scope: !1, file: !1, type: !4, spFlags: DISPFlagLocalToUnit | DISPFlagDefinition | DISPFlagOptimized, unit: !0, retainedNodes: !2452)
conflicting debug info for argument
  call void @llvm.dbg.declare(metadata i8** %7, metadata !2453, metadata !DIExpression()), !dbg !2450
!2443 = !DILocalVariable(name: "arg1", arg: 1, scope: !2441, file: !1, type: !7)
!2453 = !DILocalVariable(name: "arg1", arg: 1, scope: !2451, file: !1, type: !7)
conflicting debug info for argument
  call void @llvm.dbg.declare(metadata i32** %8, metadata !2454, metadata !DIExpression()), !dbg !2450
!2444 = !DILocalVariable(name: "arg2", arg: 2, scope: !2441, file: !1, type: !8)
!2454 = !DILocalVariable(name: "arg2", arg: 2, scope: !2451, file: !1, type: !8)
conflicting debug info for argument
  call void @llvm.dbg.declare(metadata i32* %9, metadata !2455, metadata !DIExpression()), !dbg !2450
!2445 = !DILocalVariable(name: "arg3", arg: 3, scope: !2441, file: !1, type: !6)
!2455 = !DILocalVariable(name: "arg3", arg: 3, scope: !2451, file: !1, type: !6)
conflicting debug info for argument
  call void @llvm.dbg.declare(metadata i8** %10, metadata !2456, metadata !DIExpression()), !dbg !2450
!2446 = !DILocalVariable(name: "arg4", arg: 4, scope: !2441, file: !1, type: !7)
!2456 = !DILocalVariable(name: "arg4", arg: 4, scope: !2451, file: !1, type: !7)
conflicting debug info for argument
  call void @llvm.dbg.declare(metadata i32** %11, metadata !2457, metadata !DIExpression()), !dbg !2450
!2447 = !DILocalVariable(name: "arg5", arg: 5, scope: !2441, file: !1, type: !8)
!2457 = !DILocalVariable(name: "arg5", arg: 5, scope: !2451, file: !1, type: !8)
conflicting debug info for argument
  call void @llvm.dbg.declare(metadata i8** %12, metadata !2458, metadata !DIExpression()), !dbg !2450
!2448 = !DILocalVariable(name: "arg6", arg: 6, scope: !2441, file: !1, type: !7)
!2458 = !DILocalVariable(name: "arg6", arg: 6, scope: !2451, file: !1, type: !7)
conflicting debug info for argument
  call void @llvm.dbg.declare(metadata i8** %14, metadata !2462, metadata !DIExpression()), !dbg !2459
!2453 = !DILocalVariable(name: "arg1", arg: 1, scope: !2451, file: !1, type: !7)
!2462 = !DILocalVariable(name: "arg1", arg: 1, scope: !2460, file: !1, type: !7)
conflicting debug info for argument
  call void @llvm.dbg.declare(metadata i32** %15, metadata !2463, metadata !DIExpression()), !dbg !2459
!2454 = !DILocalVariable(name: "arg2", arg: 2, scope: !2451, file: !1, type: !8)
!2463 = !DILocalVariable(name: "arg2", arg: 2, scope: !2460, file: !1, type: !8)
conflicting debug info for argument
  call void @llvm.dbg.declare(metadata i32* %16, metadata !2464, metadata !DIExpression()), !dbg !2459
!2455 = !DILocalVariable(name: "arg3", arg: 3, scope: !2451, file: !1, type: !6)
!2464 = !DILocalVariable(name: "arg3", arg: 3, scope: !2460, file: !1, type: !6)
conflicting debug info for argument
  call void @llvm.dbg.declare(metadata i8** %17, metadata !2465, metadata !DIExpression()), !dbg !2459
!2456 = !DILocalVariable(name: "arg4", arg: 4, scope: !2451, file: !1, type: !7)
!2465 = !DILocalVariable(name: "arg4", arg: 4, scope: !2460, file: !1, type: !7)
conflicting debug info for argument
  call void @llvm.dbg.declare(metadata i32** %18, metadata !2466, metadata !DIExpression()), !dbg !2459
!2457 = !DILocalVariable(name: "arg5", arg: 5, scope: !2451, file: !1, type: !8)
!2466 = !DILocalVariable(name: "arg5", arg: 5, scope: !2460, file: !1, type: !8)
conflicting debug info for argument
  call void @llvm.dbg.declare(metadata i8** %19, metadata !2467, metadata !DIExpression()), !dbg !2459
!2458 = !DILocalVariable(name: "arg6", arg: 6, scope: !2451, file: !1, type: !7)
!2467 = !DILocalVariable(name: "arg6", arg: 6, scope: !2460, file: !1, type: !7)
conflicting debug info for argument
  call void @llvm.dbg.declare(metadata i8** %21, metadata !2471, metadata !DIExpression()), !dbg !2468
!2462 = !DILocalVariable(name: "arg1", arg: 1, scope: !2460, file: !1, type: !7)
!2471 = !DILocalVariable(name: "arg1", arg: 1, scope: !2469, file: !1, type: !7)
conflicting debug info for argument
  call void @llvm.dbg.declare(metadata i32** %22, metadata !2472, metadata !DIExpression()), !dbg !2468
!2463 = !DILocalVariable(name: "arg2", arg: 2, scope: !2460, file: !1, type: !8)
!2472 = !DILocalVariable(name: "arg2", arg: 2, scope: !2469, file: !1, type: !8)
conflicting debug info for argument
  call void @llvm.dbg.declare(metadata i32* %23, metadata !2473, metadata !DIExpression()), !dbg !2468
!2464 = !DILocalVariable(name: "arg3", arg: 3, scope: !2460, file: !1, type: !6)
!2473 = !DILocalVariable(name: "arg3", arg: 3, scope: !2469, file: !1, type: !6)
conflicting debug info for argument
  call void @llvm.dbg.declare(metadata i8** %24, metadata !2474, metadata !DIExpression()), !dbg !2468
!2465 = !DILocalVariable(name: "arg4", arg: 4, scope: !2460, file: !1, type: !7)
!2474 = !DILocalVariable(name: "arg4", arg: 4, scope: !2469, file: !1, type: !7)
conflicting debug info for argument
  call void @llvm.dbg.declare(metadata i32** %25, metadata !2475, metadata !DIExpression()), !dbg !2468
!2466 = !DILocalVariable(name: "arg5", arg: 5, scope: !2460, file: !1, type: !8)
!2475 = !DILocalVariable(name: "arg5", arg: 5, scope: !2469, file: !1, type: !8)
conflicting debug info for argument
  call void @llvm.dbg.declare(metadata i8** %26, metadata !2476, metadata !DIExpression()), !dbg !2468
!2467 = !DILocalVariable(name: "arg6", arg: 6, scope: !2460, file: !1, type: !7)
!2476 = !DILocalVariable(name: "arg6", arg: 6, scope: !2469, file: !1, type: !7)
conflicting debug info for argument
  call void @llvm.dbg.declare(metadata i8** %28, metadata !2480, metadata !DIExpression()), !dbg !2477
!2471 = !DILocalVariable(name: "arg1", arg: 1, scope: !2469, file: !1, type: !7)
!2480 = !DILocalVariable(name: "arg1", arg: 1, scope: !2478, file: !1, type: !7)
conflicting debug info for argument
  call void @llvm.dbg.declare(metadata i32** %29, metadata !2481, metadata !DIExpression()), !dbg !2477
!2472 = !DILocalVariable(name: "arg2", arg: 2, scope: !2469, file: !1, type: !8)
!2481 = !DILocalVariable(name: "arg2", arg: 2, scope: !2478, file: !1, type: !8)
conflicting debug info for argument
  call void @llvm.dbg.declare(metadata i32* %30, metadata !2482, metadata !DIExpression()), !dbg !2477
!2473 = !DILocalVariable(name: "arg3", arg: 3, scope: !2469, file: !1, type: !6)
!2482 = !DILocalVariable(name: "arg3", arg: 3, scope: !2478, file: !1, type: !6)
conflicting debug info for argument
  call void @llvm.dbg.declare(metadata i8** %31, metadata !2483, metadata !DIExpression()), !dbg !2477
!2474 = !DILocalVariable(name: "arg4", arg: 4, scope: !2469, file: !1, type: !7)
!2483 = !DILocalVariable(name: "arg4", arg: 4, scope: !2478, file: !1, type: !7)
conflicting debug info for argument
  call void @llvm.dbg.declare(metadata i32** %32, metadata !2484, metadata !DIExpression()), !dbg !2477
!2475 = !DILocalVariable(name: "arg5", arg: 5, scope: !2469, file: !1, type: !8)
!2484 = !DILocalVariable(name: "arg5", arg: 5, scope: !2478, file: !1, type: !8)
conflicting debug info for argument
  call void @llvm.dbg.declare(metadata i8** %33, metadata !2485, metadata !DIExpression()), !dbg !2477
!2476 = !DILocalVariable(name: "arg6", arg: 6, scope: !2469, file: !1, type: !7)
!2485 = !DILocalVariable(name: "arg6", arg: 6, scope: !2478, file: !1, type: !7)
conflicting debug info for argument
  call void @llvm.dbg.declare(metadata i8** %35, metadata !2489, metadata !DIExpression()), !dbg !2486
!2480 = !DILocalVariable(name: "arg1", arg: 1, scope: !2478, file: !1, type: !7)
!2489 = !DILocalVariable(name: "arg1", arg: 1, scope: !2487, file: !1, type: !7)
conflicting debug info for argument
  call void @llvm.dbg.declare(metadata i32** %36, metadata !2490, metadata !DIExpression()), !dbg !2486
!2481 = !DILocalVariable(name: "arg2", arg: 2, scope: !2478, file: !1, type: !8)
!2490 = !DILocalVariable(name: "arg2", arg: 2, scope: !2487, file: !1, type: !8)
conflicting debug info for argument
  call void @llvm.dbg.declare(metadata i32* %37, metadata !2491, metadata !DIExpression()), !dbg !2486
!2482 = !DILocalVariable(name: "arg3", arg: 3, scope: !2478, file: !1, type: !6)
!2491 = !DILocalVariable(name: "arg3", arg: 3, scope: !2487, file: !1, type: !6)
conflicting debug info for argument
  call void @llvm.dbg.declare(metadata i8** %38, metadata !2492, metadata !DIExpression()), !dbg !2486
!2483 = !DILocalVariable(name: "arg4", arg: 4, scope: !2478, file: !1, type: !7)
!2492 = !DILocalVariable(name: "arg4", arg: 4, scope: !2487, file: !1, type: !7)
conflicting debug info for argument
  call void @llvm.dbg.declare(metadata i32** %39, metadata !2493, metadata !DIExpression()), !dbg !2486
!2484 = !DILocalVariable(name: "arg5", arg: 5, scope: !2478, file: !1, type: !8)
!2493 = !DILocalVariable(name: "arg5", arg: 5, scope: !2487, file: !1, type: !8)
conflicting debug info for argument
  call void @llvm.dbg.declare(metadata i8** %40, metadata !2494, metadata !DIExpression()), !dbg !2486
!2485 = !DILocalVariable(name: "arg6", arg: 6, scope: !2478, file: !1, type: !7)
!2494 = !DILocalVariable(name: "arg6", arg: 6, scope: !2487, file: !1, type: !7)
conflicting debug info for argument
  call void @llvm.dbg.declare(metadata i8** %42, metadata !2498, metadata !DIExpression()), !dbg !2495
!2489 = !DILocalVariable(name: "arg1", arg: 1, scope: !2487, file: !1, type: !7)
!2498 = !DILocalVariable(name: "arg1", arg: 1, scope: !2496, file: !1, type: !7)
conflicting debug info for argument
  call void @llvm.dbg.declare(metadata i32** %43, metadata !2499, metadata !DIExpression()), !dbg !2495
!2490 = !DILocalVariable(name: "arg2", arg: 2, scope: !2487, file: !1, type: !8)
!2499 = !DILocalVariable(name: "arg2", arg: 2, scope: !2496, file: !1, type: !8)
conflicting debug info for argument
  call void @llvm.dbg.declare(metadata i32* %44, metadata !2500, metadata !DIExpression()), !dbg !2495
!2491 = !DILocalVariable(name: "arg3", arg: 3, scope: !2487, file: !1, type: !6)
!2500 = !DILocalVariable(name: "arg3", arg: 3, scope: !2496, file: !1, type: !6)
conflicting debug info for argument
  call void @llvm.dbg.declare(metadata i8** %45, metadata !2501, metadata !DIExpression()), !dbg !2495
!2492 = !DILocalVariable(name: "arg4", arg: 4, scope: !2487, file: !1, type: !7)
!2501 = !DILocalVariable(name: "arg4", arg: 4, scope: !2496, file: !1, type: !7)
conflicting debug info for argument
  call void @llvm.dbg.declare(metadata i32** %46, metadata !2502, metadata !DIExpression()), !dbg !2495
!2493 = !DILocalVariable(name: "arg5", arg: 5, scope: !2487, file: !1, type: !8)
!2502 = !DILocalVariable(name: "arg5", arg: 5, scope: !2496, file: !1, type: !8)
conflicting debug info for argument
  call void @llvm.dbg.declare(metadata i8** %47, metadata !2503, metadata !DIExpression()), !dbg !2495
!2494 = !DILocalVariable(name: "arg6", arg: 6, scope: !2487, file: !1, type: !7)
!2503 = !DILocalVariable(name: "arg6", arg: 6, scope: !2496, file: !1, type: !7)
```

Has anyone met this problem?

**Comment by chapados** [comment]
I ran into a similar error attempting to build for an `iphone` target.  I was using the prebuilt tvm (from `pip3 install mlc-ai-nightly -f https://mlc.ai/wheels`). To get past this, I ended up building tvm from source based on the instructions in the [ios/README](https://github.com/mlc-ai/mlc-llm/blob/main/ios/README.md).  I'm building on an M1 mac and needed to install llvmdev through conda (`conda install llvmdev`).
